Orange/data/domain.py

```diff
@@ -254,6 +258,13 @@ def empty(self):
         """True if the domain has no variables of any kind"""
         return not self.variables and not self.metas
 
+    def _get_equivalent(self, var):
+        if isinstance(var, Variable):
```

Comment:
I tried this:

    import time
    import Orange
    
    data = Orange.data.Table([list(range(5000))])
    
    t = time.time()
    pdata = Orange.preprocess.SklImpute()(data)
    print("first transform", time.time()-t, "s")
    
    t = time.time()
    data.transform(pdata.domain)
    print("Table.tranform", time.time()-t, "s")

On master I get 

    first transform 0.6564505100250244 s
    Table.tranform 0.5023694038391113 s

but on this branch

    first transform 11.536584615707397 s
    Table.tranform 0.5088396072387695 s

My profiler led me here.